src/vcpkg/build.cpp

```diff
@@ -309,7 +309,26 @@ namespace vcpkg::Build
         "",
         "Error: in triplet {triplet}: Unable to find a valid toolchain combination.\n    The requested target "
         "architecture was {value}\n    "
-        "The selected Visual Studio instance is at {path}\n    The available toolchain combinations are {list}\n");
+        "The selected Visual Studio instance is at {path}\n    The available toolchain combinations are {list}\n"
+        "Please check the required Visual Studio Components:\n"
+        "C++:\n"
+        "   Windows Universal C Runtime\n"
+        "   C++ core desktop features\n"
+        "   One of the following Visual Studio Build tools:\n"
+        "       * Microsoft.VisualStudio.Component.VC.140 (for Visual Studio 2015)\n"
+        "       * MSVC v141/v142/v143 - VS 2017/2019/2022 C++ x64/x86 build tools (for Visual Studio 2017 or later)\n"
+        "   MSBuild\n"
+        "One of the following Windows SDKs:\n"
+        "   * 8.1\n"
+        "   * 10.0.18362\n"
+        "   * 10.0.19041\n"
+        "   * 10.0.20348\n"
+        "   * 11.0.22000\n"
```

Comment:
This seems fundamentally problematic. We shouldn't need to publish a new version of vcpkg for every Windows SDK update, but that's what's implied by listing out 5 specific versions. Why can this not simply be "Windows SDK 8.1 or greater"?

Reply:
In fact, using `Microsoft.VisualStudio.Component.Windows10SDK` to check the Windows SDK will not work, even if it is not installed it will return the current Visual Studio version.